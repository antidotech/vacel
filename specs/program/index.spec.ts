import { runSpec } from '../__helper__'

runSpec(__dirname, {})
