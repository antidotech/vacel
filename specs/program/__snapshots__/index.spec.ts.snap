// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`program single_handling ast: ast 1`] = `
Node {
  "body": Array [
    Node {
      "body": Array [
        Node {
          "id": Node {
            "base": Node {
              "loc": Object {
                "end": Object {
                  "column": 19,
                  "line": 2,
                  "offset": 33,
                },
                "start": Object {
                  "column": 17,
                  "line": 2,
                  "offset": 31,
                },
              },
              "name": "var",
              "type": "Identifier",
            },
            "loc": Object {
              "end": Object {
                "column": 32,
                "line": 2,
                "offset": 46,
              },
              "start": Object {
                "column": 17,
                "line": 2,
                "offset": 31,
              },
            },
            "member": Node {
              "loc": Object {
                "end": Object {
                  "column": 32,
                  "line": 2,
                  "offset": 46,
                },
                "start": Object {
                  "column": 21,
                  "line": 2,
                  "offset": 35,
                },
              },
              "name": "vacelinDebug",
              "type": "Identifier",
            },
            "type": "Member",
          },
          "loc": Object {
            "end": Object {
              "column": 40,
              "line": 2,
              "offset": 54,
            },
            "start": Object {
              "column": 3,
              "line": 2,
              "offset": 17,
            },
          },
          "type": "DeclareStatement",
          "valueType": "STRING",
        },
        Node {
          "left": Node {
            "base": Node {
              "loc": Object {
                "end": Object {
                  "column": 9,
                  "line": 3,
                  "offset": 64,
                },
                "start": Object {
                  "column": 7,
                  "line": 3,
                  "offset": 62,
                },
              },
              "name": "var",
              "type": "Identifier",
            },
            "loc": Object {
              "end": Object {
                "column": 23,
                "line": 3,
                "offset": 78,
              },
              "start": Object {
                "column": 7,
                "line": 3,
                "offset": 62,
              },
            },
            "member": Node {
              "loc": Object {
                "end": Object {
                  "column": 23,
                  "line": 3,
                  "offset": 78,
                },
                "start": Object {
                  "column": 11,
                  "line": 3,
                  "offset": 66,
                },
              },
              "name": "vacelineDebug",
              "type": "Identifier",
            },
            "type": "Member",
          },
          "loc": Object {
            "end": Object {
              "column": 36,
              "line": 3,
              "offset": 91,
            },
            "start": Object {
              "column": 3,
              "line": 3,
              "offset": 58,
            },
          },
          "operator": "=",
          "right": Node {
            "loc": Object {
              "end": Object {
                "column": 35,
                "line": 3,
                "offset": 90,
              },
              "start": Object {
                "column": 27,
                "line": 3,
                "offset": 82,
              },
            },
            "type": "StringLiteral",
            "value": "\\"initial\\"",
          },
          "type": "SetStatement",
        },
        Node {
          "alternative": undefined,
          "consequent": Array [
            Node {
              "left": Node {
                "base": Node {
                  "loc": Object {
                    "end": Object {
                      "column": 11,
                      "line": 6,
                      "offset": 145,
                    },
                    "start": Object {
                      "column": 9,
                      "line": 6,
                      "offset": 143,
                    },
                  },
                  "name": "var",
                  "type": "Identifier",
                },
                "loc": Object {
                  "end": Object {
                    "column": 24,
                    "line": 6,
                    "offset": 158,
                  },
                  "start": Object {
                    "column": 9,
                    "line": 6,
                    "offset": 143,
                  },
                },
                "member": Node {
                  "loc": Object {
                    "end": Object {
                      "column": 24,
                      "line": 6,
                      "offset": 158,
                    },
                    "start": Object {
                      "column": 13,
                      "line": 6,
                      "offset": 147,
                    },
                  },
                  "name": "vacelinDebug",
                  "type": "Identifier",
                },
                "type": "Member",
              },
              "loc": Object {
                "end": Object {
                  "column": 47,
                  "line": 6,
                  "offset": 181,
                },
                "start": Object {
                  "column": 5,
                  "line": 6,
                  "offset": 139,
                },
              },
              "operator": "=",
              "right": Node {
                "loc": Object {
                  "end": Object {
                    "column": 46,
                    "line": 6,
                    "offset": 180,
                  },
                  "start": Object {
                    "column": 28,
                    "line": 6,
                    "offset": 162,
                  },
                },
                "type": "StringLiteral",
                "value": "\\"Vacelin Debug Log\\"",
              },
              "type": "AddStatement",
            },
          ],
          "loc": Object {
            "end": Object {
              "column": 3,
              "line": 7,
              "offset": 185,
            },
            "start": Object {
              "column": 3,
              "line": 5,
              "offset": 96,
            },
          },
          "test": Node {
            "left": Node {
              "base": Node {
                "base": Node {
                  "base": Node {
                    "loc": Object {
                      "end": Object {
                        "column": 9,
                        "line": 5,
                        "offset": 102,
                      },
                      "start": Object {
                        "column": 7,
                        "line": 5,
                        "offset": 100,
                      },
                    },
                    "name": "req",
                    "type": "Identifier",
                  },
                  "loc": Object {
                    "end": Object {
                      "column": 14,
                      "line": 5,
                      "offset": 107,
                    },
                    "start": Object {
                      "column": 7,
                      "line": 5,
                      "offset": 100,
                    },
                  },
                  "member": Node {
                    "loc": Object {
                      "end": Object {
                        "column": 14,
                        "line": 5,
                        "offset": 107,
                      },
                      "start": Object {
                        "column": 11,
                        "line": 5,
                        "offset": 104,
                      },
                    },
                    "name": "http",
                    "type": "Identifier",
                  },
                  "type": "Member",
                },
                "loc": Object {
                  "end": Object {
                    "column": 22,
                    "line": 5,
                    "offset": 115,
                  },
                  "start": Object {
                    "column": 7,
                    "line": 5,
                    "offset": 100,
                  },
                },
                "member": Node {
                  "loc": Object {
                    "end": Object {
                      "column": 22,
                      "line": 5,
                      "offset": 115,
                    },
                    "start": Object {
                      "column": 16,
                      "line": 5,
                      "offset": 109,
                    },
                  },
                  "name": "Vacelin",
                  "type": "Identifier",
                },
                "type": "Member",
              },
              "loc": Object {
                "end": Object {
                  "column": 27,
                  "line": 5,
                  "offset": 120,
                },
                "start": Object {
                  "column": 7,
                  "line": 5,
                  "offset": 100,
                },
              },
              "name": Node {
                "loc": Object {
                  "end": Object {
                    "column": 27,
                    "line": 5,
                    "offset": 120,
                  },
                  "start": Object {
                    "column": 24,
                    "line": 5,
                    "offset": 117,
                  },
                },
                "name": "Some",
                "type": "Identifier",
              },
              "type": "ValuePair",
            },
            "loc": Object {
              "end": Object {
                "column": 37,
                "line": 5,
                "offset": 130,
              },
              "start": Object {
                "column": 7,
                "line": 5,
                "offset": 100,
              },
            },
            "operator": "==",
            "right": Node {
              "loc": Object {
                "end": Object {
                  "column": 37,
                  "line": 5,
                  "offset": 130,
                },
                "start": Object {
                  "column": 32,
                  "line": 5,
                  "offset": 125,
                },
              },
              "type": "StringLiteral",
              "value": "\\"Some\\"",
            },
            "type": "BinaryExpression",
          },
          "type": "IfStatement",
        },
      ],
      "id": Node {
        "loc": Object {
          "end": Object {
            "column": 12,
            "line": 1,
            "offset": 11,
          },
          "start": Object {
            "column": 5,
            "line": 1,
            "offset": 4,
          },
        },
        "name": "vcl_recv",
        "type": "Identifier",
      },
      "loc": Object {
        "end": Object {
          "column": 1,
          "line": 8,
          "offset": 187,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "SubroutineStatement",
    },
  ],
  "loc": Object {
    "end": Object {
      "column": 1,
      "line": 8,
      "offset": 187,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "Program",
}
`;

exports[`program single_handling format: format long 1`] = `
sub vcl_recv {
  declare local var.vacelinDebug STRING;
  set var.vacelineDebug = "initial";

  if (req.http.Vacelin:Some == "Some") {
    add var.vacelinDebug = "Vacelin Debug Log";
  }
}
`;

exports[`program single_handling format: format short 1`] = `
sub vcl_recv {
  declare local var.vacelinDebug STRING;
  set var.vacelineDebug =
    "initial";

  if (
    req
      .http
      .Vacelin:Some ==
      "Some"
  ) {
    add var.vacelinDebug =
      "Vacelin Debug Log";
  }
}
`;

exports[`program single_handling token 1`] = `
Array [
  sub,
  vcl_recv,
  {,
  declare,
  local,
  var,
  .,
  vacelinDebug,
  STRING,
  ;,
  set,
  var,
  .,
  vacelineDebug,
  =,
  "initial",
  ;,
  if,
  (,
  req,
  .,
  http,
  .,
  Vacelin,
  :,
  Some,
  ==,
  "Some",
  ),
  {,
  add,
  var,
  .,
  vacelinDebug,
  =,
  "Vacelin Debug Log",
  ;,
  },
  },
]
`;

exports[`program with_empty_lines ast: ast 1`] = `
Node {
  "body": Array [
    Node {
      "id": Node {
        "base": Node {
          "loc": Object {
            "end": Object {
              "column": 17,
              "line": 1,
              "offset": 16,
            },
            "start": Object {
              "column": 15,
              "line": 1,
              "offset": 14,
            },
          },
          "name": "var",
          "type": "Identifier",
        },
        "loc": Object {
          "end": Object {
            "column": 30,
            "line": 1,
            "offset": 29,
          },
          "start": Object {
            "column": 15,
            "line": 1,
            "offset": 14,
          },
        },
        "member": Node {
          "loc": Object {
            "end": Object {
              "column": 30,
              "line": 1,
              "offset": 29,
            },
            "start": Object {
              "column": 19,
              "line": 1,
              "offset": 18,
            },
          },
          "name": "vacelinDebug",
          "type": "Identifier",
        },
        "type": "Member",
      },
      "loc": Object {
        "end": Object {
          "column": 38,
          "line": 1,
          "offset": 37,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "DeclareStatement",
      "valueType": "STRING",
    },
    Node {
      "left": Node {
        "base": Node {
          "loc": Object {
            "end": Object {
              "column": 7,
              "line": 2,
              "offset": 45,
            },
            "start": Object {
              "column": 5,
              "line": 2,
              "offset": 43,
            },
          },
          "name": "var",
          "type": "Identifier",
        },
        "loc": Object {
          "end": Object {
            "column": 21,
            "line": 2,
            "offset": 59,
          },
          "start": Object {
            "column": 5,
            "line": 2,
            "offset": 43,
          },
        },
        "member": Node {
          "loc": Object {
            "end": Object {
              "column": 21,
              "line": 2,
              "offset": 59,
            },
            "start": Object {
              "column": 9,
              "line": 2,
              "offset": 47,
            },
          },
          "name": "vacelineDebug",
          "type": "Identifier",
        },
        "type": "Member",
      },
      "loc": Object {
        "end": Object {
          "column": 28,
          "line": 2,
          "offset": 66,
        },
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 39,
        },
      },
      "operator": "=",
      "right": Node {
        "loc": Object {
          "end": Object {
            "column": 27,
            "line": 2,
            "offset": 65,
          },
          "start": Object {
            "column": 25,
            "line": 2,
            "offset": 63,
          },
        },
        "type": "StringLiteral",
        "value": "\\"1\\"",
      },
      "type": "SetStatement",
    },
    Node {
      "left": Node {
        "base": Node {
          "loc": Object {
            "end": Object {
              "column": 7,
              "line": 4,
              "offset": 75,
            },
            "start": Object {
              "column": 5,
              "line": 4,
              "offset": 73,
            },
          },
          "name": "var",
          "type": "Identifier",
        },
        "loc": Object {
          "end": Object {
            "column": 21,
            "line": 4,
            "offset": 89,
          },
          "start": Object {
            "column": 5,
            "line": 4,
            "offset": 73,
          },
        },
        "member": Node {
          "loc": Object {
            "end": Object {
              "column": 21,
              "line": 4,
              "offset": 89,
            },
            "start": Object {
              "column": 9,
              "line": 4,
              "offset": 77,
            },
          },
          "name": "vacelineDebug",
          "type": "Identifier",
        },
        "type": "Member",
      },
      "loc": Object {
        "end": Object {
          "column": 28,
          "line": 4,
          "offset": 96,
        },
        "start": Object {
          "column": 1,
          "line": 4,
          "offset": 69,
        },
      },
      "operator": "=",
      "right": Node {
        "loc": Object {
          "end": Object {
            "column": 27,
            "line": 4,
            "offset": 95,
          },
          "start": Object {
            "column": 25,
            "line": 4,
            "offset": 93,
          },
        },
        "type": "StringLiteral",
        "value": "\\"2\\"",
      },
      "type": "SetStatement",
    },
  ],
  "loc": Object {
    "end": Object {
      "column": 28,
      "line": 4,
      "offset": 96,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "Program",
}
`;

exports[`program with_empty_lines format: format long 1`] = `
declare local var.vacelinDebug STRING;
set var.vacelineDebug = "1";

set var.vacelineDebug = "2";
`;

exports[`program with_empty_lines format: format short 1`] = `
declare local var.vacelinDebug STRING;
set var.vacelineDebug =
  "1";

set var.vacelineDebug =
  "2";
`;

exports[`program with_empty_lines token 1`] = `
Array [
  declare,
  local,
  var,
  .,
  vacelinDebug,
  STRING,
  ;,
  set,
  var,
  .,
  vacelineDebug,
  =,
  "1",
  ;,
  set,
  var,
  .,
  vacelineDebug,
  =,
  "2",
  ;,
]
`;
